source/Calamari/Kubernetes/Integration/Kubectl.cs

```diff
@@ -5,14 +5,22 @@
 using Calamari.Common.Features.Processes;
 using Calamari.Common.Plumbing;
 using Calamari.Common.Plumbing.Logging;
-using Calamari.Common.Plumbing.Variables;
 using Newtonsoft.Json;
 using Octopus.CoreUtilities;
 using Octopus.Versioning.Semver;
 
 namespace Calamari.Kubernetes.Integration
 {
-    public class Kubectl : CommandLineTool
+    public interface IKubectl
```

Comment:
This interface was added because I wanted to swap the implementation with a mock in the tests, and at the same time I wanted to rely on the existing `Kubectl` implementation which contains the methods to locate custom kubectl tool etc.

Thinking about it now, it would nicer to have a smaller interface that is only used in the status check part (maybe called `IKubectlGet`) which takes the existing `Kubectl` tool as a dependency. This will make the mock implementation much easier to write as well.

Reply:
Just updated